logr.go

```diff
@@ -33,16 +34,16 @@ type Logger interface {
 	// example, logger.Info() produces the same result as logger.V(0).Info.
 	InfoLogger
 
-	// Error logs a error message.  This is behaviorally akin to fmt.Print.
-	Error(args ...interface{})
-
-	// Errorf logs a formatted error message.
-	Errorf(format string, args ...interface{})
+	// Error logs an error, with the given message and key/value pairs as context.
+	Error(err error, msg string, keysAndValues ...interface{})
```

Comment:
What is the intended distinction between the `err` argument and the `msg` ?

Reply:
msg is the "context" around the error, err is attached as structured logging.  It's nice wrapping around the common error logging pattern that looks like `glog.Errorf("Failed to frob the bits: %v", err)` in non-structured form.  Will document better.

Reply:
(more concretely, things like zap key on msg for sampling as well, and sometimes errors hide more complexity than implied by the error interface, like `AggregateError`, which zap handles natively, or things like `StatusError`)